File: python/p9.py
def pythagorean_triplet(total):
	return [x*y*((x**2 + y**2)**0.5)
				for y in xrange(2, total)
				for x in xrange(1, y) if x+y+((x**2 + y**2)**0.5) == total]

# Product of sides of pythagorean triplet for which sum of sides is 1000
print pythagorean_triplet(1000)

File: javascript/p9.js
const pythagorean_triplet = (tot) => {
	let y = [...Array(tot-1).keys()].map((n) => { return n + 1; });


	for (let yi of y) {
		for (let xi of [...Array(yi).keys()].map((n) => { return n + 1; })) {
			let z = Math.pow((Math.pow(xi, 2) + Math.pow(yi, 2)), 0.5);
			if (xi + yi + z === tot) { return xi*yi*z; }
		}
	}
}

// Product of sides of pythagorean triplet for which sum of sides is 1000
console.log(pythagorean_triplet(1000))